src/components/BannerConductor/BannerConductor.vue

```diff
@@ -0,0 +1,62 @@
+<template>
+	<div ref="bannerRef" class="wmde-banner" :class="{
+		'wmde-banner--pending' : bannerState === BannerState.Pending,
+		'wmde-banner--showing' : bannerState === BannerState.Showing,
+		'wmde-banner--visible' : bannerState === BannerState.Visible,
+		'wmde-banner--closed' : bannerState === BannerState.Closed
+	}">
+		<component :is="banner" v-bind="bannerProps" :bannerState="bannerState"/>
+	</div>
+</template>
+
+<script setup lang="ts">
+
+import { Page } from '@src/page/Page';
+import { onMounted, ref } from 'vue';
+import { BannerConfig } from '@src/BannerConfig';
+import { BannerState } from '@src/components/BannerConductor/BannerState';
+import { ResizeHandler } from '@src/utils/ResizeHandler';
+
+interface Props {
+	page: Page,
+	bannerConfig: BannerConfig,
+	resizeHandler: ResizeHandler,
+	banner: Object,
+	bannerProps: Object
+}
+
+const props = withDefaults( defineProps<Props>(), {} );
+const bannerRef = ref( null );
+const bannerState = ref<BannerState>( BannerState.Pending );
+
+onMounted( () => {
+	props.page.setTransitionDuration( props.bannerConfig.transitionDuration )
+		.setSpace( bannerRef.value.offsetHeight );
+
+	setTimeout( () => {
+		bannerState.value = BannerState.Showing;
+		props.page.setAnimated()
+			.showBanner();
```

Comment:
From the `BannerConductor` name I guess you'll add more code here?

At the end of the timeout it should
1. Check if banner should/can be displayed
2. Do some event tracking (if it was not displayed, log reason (size issues,  etc) and notify CN see https://phabricator.wikimedia.org/T310982 )
3. set animated/shown properties if it should be displayed.